src/test/java/io/jenkins/plugins/ml/IPythonServerGlobalConfigurationTest.java

```diff
@@ -0,0 +1,120 @@
+import java.util.List;
+
+import static junit.framework.TestCase.assertEquals;
+
+public class IPythonServerGlobalConfigurationTest {
+
+    private HtmlForm form;
+    private HtmlPage configPage;
+
+    @Rule
+    public final JenkinsRule j = new JenkinsRule();
+
+    @Before
+    public void setup() throws IOException, SAXException {
+        configPage = j.createWebClient().goTo("configure");
+        form = configPage.getFormByName("config");
+        getButton(form, "Add new Server")
+                .click();
+        getButton(form, "Add new Server")
+                .click();
+    }
+
+    @Test
+    public void reloadConfigPageTest() throws Exception {
+
+        List<HtmlInput> serverName = form.getInputsByName("_.serverName");
+        serverName.get(0).setValueAttribute("localHost");
+        serverName.get(1).setValueAttribute("IPython_Server");
+        List<HtmlInput> serverAddress = form.getInputsByName("_.serverAddress");
+        serverAddress.get(0).setValueAttribute("127.0.0.1");
+        serverAddress.get(1).setValueAttribute("192.168.12.1");
+        List<HtmlInput> launchTimeout = form.getInputsByName("_.launchTimeout");
+        launchTimeout.get(0).setValueAttribute("5");
+        launchTimeout.get(1).setValueAttribute("1");
+        List<HtmlInput> maxResults = form.getInputsByName("_.maxResults");
+        maxResults.get(0).setValueAttribute("3");
+        maxResults.get(1).setValueAttribute("1");
+
+        j.submit(form);
+
+        form = configPage.getFormByName("config");
+        serverName = form.getInputsByName("_.serverName");
+        serverAddress = form.getInputsByName("_.serverAddress");
+        launchTimeout = form.getInputsByName("_.launchTimeout");
+        maxResults= form.getInputsByName("_.maxResults");
+
+        assertEquals("localHost", serverName.get(0).getValueAttribute());
+        assertEquals("127.0.0.1", serverAddress.get(0).getValueAttribute());
+        assertEquals("5", launchTimeout.get(0).getValueAttribute());
+        assertEquals("3", maxResults.get(0).getValueAttribute());
+
+        assertEquals("IPython_Server", serverName.get(1).getValueAttribute());
+        assertEquals("192.168.12.1", serverAddress.get(1).getValueAttribute());
+        assertEquals("1", launchTimeout.get(1).getValueAttribute());
+        assertEquals("1", maxResults.get(1).getValueAttribute());
+
+        configPage.refresh();
+
+        assertEquals("localHost", serverName.get(0).getValueAttribute());
+        assertEquals("127.0.0.1", serverAddress.get(0).getValueAttribute());
+        assertEquals("5", launchTimeout.get(0).getValueAttribute());
+        assertEquals("3", maxResults.get(0).getValueAttribute());
+
+        assertEquals("IPython_Server", serverName.get(1).getValueAttribute());
+        assertEquals("192.168.12.1", serverAddress.get(1).getValueAttribute());
+        assertEquals("1", launchTimeout.get(1).getValueAttribute());
+        assertEquals("1", maxResults.get(1).getValueAttribute());
```

Comment:
That was super fast @Loghijiaha ! :clap:

Reply:
Yup @kinow I fell love with this automated tests using JenkinsWebClient() :heart:

Reply:
I felt the same the first time I used it! They did an amazing work with their test harness.